### PR TITLE
Integrate ldap

### PR DESCRIPTION
**Pull request by adhil0**
This PR adds a script to sync GLPI groups with LDAP groups via a user-specified YAML file. A sample YAML file as well as documentation in the README  are provided.

There are a couple LDAP python packages, but they either require >100MB of dependencies or are not that actively maintained, so I opted to use subprocess -> ldapsearch for the time being.

### CODE REVIEW

File: integration/ldap/general_ldap_example.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+Group 1: # Name of group in GLPI
+  ldap: group1-users # Name of group in LDAP
+Group 1 > Group 2:
```
Comment (dkosteck) [inline integration/ldap/general_ldap_example.yaml:4]: Why use the angle bracket here, just for a newline? Does this mean both Group 1 and Group 2 of GLPI are mapped to group2-users in LDAP?

**Review by dkosteck — APPROVED** [review]
Overall I think this is a good addition, thanks. I just added one clarifying comment

File: integration/ldap/general_ldap_example.yaml (hunk 1 above)
Comment (adhil0) [reply integration/ldap/general_ldap_example.yaml:8]: This is how descendant groups are denoted in GLPI. So in this example group 2 is a subset of group 1, and group2-users will be mapped to only Group 2 of GLPI. I've added a clarifying comment to the YAML

Comment (dkosteck) [reply integration/ldap/general_ldap_example.yaml:8]: Got it, makes sense. Yeah I think the clarification is good

**Pull request merged**